### PR TITLE
📦 NEW: Add terms enum query for fetching autocompletion terms match

### PR DESCRIPTION
**Pull request by michaelborn**
Add support for a terms enum retrieval:

```js
var terms = getInstance( "HyperClient@cbElasticsearch" )
            .getTermsEnum( "hotels", {
                "field" : "city",
                "string" : "alb"
            } );
```

https://www.elastic.co/guide/en/elasticsearch/reference/8.7/search-terms-enum.html

- [x] Added `hyperClient.getTermsEnum()` 🚀
- [x] Added docs 📖
- [x] Added tests 🤖

### CODE REVIEW

**Comment by jclausen** [comment]
I am fine with the second positional argument being a struct, which could contain a full definition, but I would prefer a more fluent method signature that matches the API:

```
function getTermsEnum( 
  required string indexName, 
  required any field,
  string match,
  numeric size=10,
  boolean caseInsensitive=true
)
```

Then if you want to pass in an opt struct, you can do so and it will just override the existing args.

**Comment by michaelborn** [comment]
Thanks @jclausen . I pushed up an update and a test for the improved syntax.

File: test-harness/tests/specs/unit/HyperClientTest.cfc [hunk 1]
```diff
@@ -928,7 +928,7 @@ component extends="coldbox.system.testing.BaseTestCase" {
 
 			} );
 
-			describe( "Post index creation tests", function(){
+			xdescribe( "Post index creation tests", function(){
```
Comment (jclausen) [inline test-harness/tests/specs/unit/HyperClientTest.cfc]: Why are these tests commented out?

File: test-harness/tests/specs/unit/HyperClientTest.cfc [hunk 2]
```diff
@@ -1149,7 +1149,7 @@ component extends="coldbox.system.testing.BaseTestCase" {
 				} );
 			});
 
-			describe( "tasks", function(){
+			xdescribe( "tasks", function(){
```
Comment (jclausen) [inline test-harness/tests/specs/unit/HyperClientTest.cfc]: Restore this test

**Review by jclausen — CHANGES_REQUESTED** [review]

File: test-harness/tests/specs/unit/HyperClientTest.cfc (hunk 2 above)
Comment (jclausen) [reply test-harness/tests/specs/unit/HyperClientTest.cfc]: Please restore all of the tests in this PR which have been `x`ed

Comment (michaelborn) [reply test-harness/tests/specs/unit/HyperClientTest.cfc]: Yikes! My bad!

**Comment by michaelborn** [comment]
Skipped tests restored.

**Pull request merged**